arc_cache.go

```diff
@@ -56,9 +80,17 @@ func (b *arccache) DeleteBlock(k cid.Cid) error {
 }
 
 func (b *arccache) Has(k cid.Cid) (bool, error) {
+	if !k.Defined() {
+		return false, ErrNotFound
+	}
+
 	if has, _, ok := b.queryCache(k); ok {
 		return has, nil
 	}
+
+	unlock := b.rLock(k)
+	defer unlock()
```

Comment:
nit: not relevent if you change the helper to `getLock` (see above) but you can do this in one line:

```go
defer b.rLock().unlock()
```

Go will defer the _last_ function call.